### PR TITLE
Add remove services for collections, objects, and related objects.

### PR DESCRIPTION
**Pull request by elrayle**
New Services:
* RemoveCollectionFromCollection
* RemoveObjectFromCollection
* RemoveRelatedObjectFromCollection
* RemoveObjectFromObject
* RemoveRelatedObjectFromObject

Updated Services:
* GetRelatedObjectsFromCollection - return array
* GetRelatedObjectsFromObject - return array

### CODE REVIEW

File: lib/hydra/pcdm/services/collection/get_related_objects.rb [hunk 1]
```diff
@@ -10,8 +10,7 @@ class GetRelatedObjectsFromCollection
 
     def self.call( parent_collection )
       raise ArgumentError, "parent_collection must be a pcdm collection" unless Hydra::PCDM.collection? parent_collection
-
-      parent_collection.related_objects
+      parent_collection.related_objects.to_a
```
Comment (jcoyne) [inline lib/hydra/pcdm/services/collection/get_related_objects.rb]: Do we need to call to_a here?  Why?  to_a can be less efficient if they only want the first element of the association.

File: lib/hydra/pcdm/services/collection/remove_collection.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      # TODO members.delete should...   (issue #103)(activefedora-aggregation issue #34)
+      #   * return child_collection when successful delete
+      #   * return nil if child_collection does not exist in parent_collection
+      #   * raise error for any other problems
+
+      # parent_collection.members.delete child_collection                                # TODO Uncomment
+
+      return nil unless Hydra::PCDM::GetCollectionsFromCollection.call( parent_collection ).include? child_collection # TODO Remove
+      removed_collection = parent_collection.members.delete child_collection             # TODO Remove
```
Comment (jcoyne) [inline lib/hydra/pcdm/services/collection/remove_collection.rb]: Please Explain the TODO here. When should that be done? Why? Why not now?

File: lib/hydra/pcdm/services/collection/remove_collection.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+      # TODO FIX when members is empty, members.delete raises ActiveFedora::ObjectNotFoundError "Can't reload an object that hasn't been saved"  (activefedora-aggregation issue #35)
+
+      # TODO members.delete should...   (issue #103)(activefedora-aggregation issue #34)
+      #   * return child_collection when successful delete
+      #   * return nil if child_collection does not exist in parent_collection
+      #   * raise error for any other problems
+
+      # parent_collection.members.delete child_collection                                # TODO Uncomment
```
Comment (jcoyne) [inline lib/hydra/pcdm/services/collection/remove_collection.rb]: Please Explain the TODO here. When should that be done? Why? Why not now?

File: lib/hydra/pcdm/services/collection/remove_collection.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+      return nil unless Hydra::PCDM::GetCollectionsFromCollection.call( parent_collection ).include? child_collection # TODO Remove
+      removed_collection = parent_collection.members.delete child_collection             # TODO Remove
+      removed_collection = removed_collection.first  if removed_collection.is_a? Array   # TODO Remove
```
Comment (jcoyne) [inline lib/hydra/pcdm/services/collection/remove_collection.rb]: Explain.

File: lib/hydra/pcdm/services/collection/remove_collection.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      removed_collection                                                                 # TODO Remove
```
Comment (jcoyne) [inline lib/hydra/pcdm/services/collection/remove_collection.rb]: Explain

File: lib/hydra/pcdm/services/collection/remove_related_object.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    # @param [Hydra::PCDM::Object] :child_related_object being removed
+    #
+    # @return [Hydra::PCDM::Collection] the updated pcdm collection
+
+    def self.call( parent_collection, child_related_object )
+      raise ArgumentError, "parent_collection must be a pcdm collection" unless Hydra::PCDM.collection? parent_collection
+      raise ArgumentError, "child_related_object must be a pcdm object" unless Hydra::PCDM.object? child_related_object
+
+
+      # TODO FIX when related_objects is empty, related_objects.delete raises ActiveFedora::ObjectNotFoundError "Can't reload an object that hasn't been saved"  (activefedora-aggregation issue #35)
```
Comment (jcoyne) [inline lib/hydra/pcdm/services/collection/remove_related_object.rb]: projecthydra-labs/activefedora-aggregation#35 is about members, not related-objects, right?

File: lib/hydra/pcdm/services/collection/remove_related_object.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+      # TODO members.delete should...   (issue #103)(activefedora-aggregation issue #34)
```
Comment (jcoyne) [inline lib/hydra/pcdm/services/collection/remove_related_object.rb]: This method doesn't operate on members, does it?

File: lib/hydra/pcdm/services/collection/get_related_objects.rb (hunk 1 above)
Comment (elrayle) [inline lib/hydra/pcdm/services/collection/get_related_objects.rb]: Two reasons for the related_object.to_a
1. related_objects.include? raises an error in some cases  (This should be fixed when issues https://github.com/projecthydra-labs/activefedora-aggregation/issues/36 is fixed)
2. col1.collections and col1.objects in collection_behaviors each returns an Array and is surfaced through services as GetCollectionsFromCollection and GetObjectsFromCollection.  For consistency in processing, col1.related_objects surfaced through GetRelatedObjectsFromCollection should return an Array.  

NOTE: Same applies for related objects in pcdm objects.

**Pull request merged**